[Daniel Kwan] iP

danielkwan2004 opened the pull request.

melchti committed:
Change name of default chatbot name, Duke, to ChatTPG. Remove all traces of Duke, and implement a skeletal version of the chatbot that simply greets the user and exits.

melchti committed:
Test commit message

This commit is to test the vim editor for the commit message

danielkwan2004 committed:
Add Level-1

Level-1 task echos commands entered by the user

danielkwan2004 committed:
Add Level-1 final with bye

Chatbot now exists when the user types the command bye

danielkwan2004 committed:
Merge branch 'master' of https://github.com/danielkwan2004/ip and Add
level-1 tasks

danielkwan2004 committed:
Add Level-2 tasks

Add ability to store whatever text entered by the user and display them
back to the user when requested.

danielkwan2004 committed:
Rename project in README.md

Updated project name from 'Duke' to 'ChatTPG' in README.

yikjin commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,19 @@
+// Add the ability to store whatever text entered by the user and display them back to the user when requested.
+public class Task {
+    
+
```
Too many blank lines (has 2, should be 1 or none)


```suggestion

```

yikjin commented on src/main/java/ChatTPG.java:
```diff
@@ -0,0 +1,27 @@
+import java.util.*;
```
Imported classes should always be added explicitly, see the [respective Java coding standard](https://se-education.org/guides/conventions/java/basic.html#package-and-import-statements)

yikjin commented on src/main/java/ChatTPG.java:
```diff
@@ -0,0 +1,27 @@
+        
+        Task task = new Task();
+
+        while (!userInput.equals("bye")) {
```
Unnecessary check here, since you already checked for this condition a few lines down


```suggestion
        while (true) {
```

yikjin commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,19 @@
+    
+
+    String[] taskList = new String[100];
+    int taskCount = 0;
```
Variable names should be plural


```suggestion
    String[] tasksList = new String[100];
    int tasksCount = 0;
```

yikjin commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,19 @@
+    String[] taskList = new String[100];
+    int taskCount = 0;
+
+    public void addTask(String description){
```
Space needed before `{`


```suggestion
    public void addTask(String description) {
```

yikjin commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,19 @@
+        System.out.println("\tadded: " + description);
+    }
+
+    public void listTask(){
```
Space needed before `{`, and method name should be plural


```suggestion
    public void listTasks() {
```

yikjin reviewed (commented):
An effort to stick to the coding style can be seen, good job! A few changes requested.

melchti committed:
Add markTaskAsDone (Level-3 task, incomplete)

danielkwan2004 committed:
Add task organiser class, add functions to manage tasks e.g. to delete a
task, mark a task as done/undone. Also add main menu

melchti committed:
Add Level-4

Add new features including different types of tasks: todo, event and
deadline. Organise code into more relevant classes and subclasses

melchti committed:
Improve code quality and ensure code is up to CS2113 coding standards

melchti committed:
Add .json files to 1-click run in vscode. Add package

melchti committed:
Add level-5

Add new features to chatbot that multiple kinds of exceptions
Change previously written edge case handling using print messages to
use exception handling

melchti committed:
Merge Level-5 into master (no fast-forward)

namsengi11 commented on src/main/java/chattpg/logic/TaskOrganiser.java:
```diff
@@ -0,0 +1,166 @@
+
+
+    public void deleteTask(int taskNumber) throws TaskIndexOutOfBoundsException {
+        if (taskNumber < 1 || taskNumber > taskCount) {
```
Good job with error handling

namsengi11 commented on src/main/java/chattpg/logic/TaskOrganiser.java:
```diff
@@ -0,0 +1,166 @@
+                throw new InvalidCommandException("event format: event <desc> /from <start> /to <end>");
+            }
+            String desc = description.substring(0, fromPos).trim();
+            String start = description.substring(fromPos + 7, toPos).trim();
```
Try to avoid use of magic numbers. At first glance, the reason for indexing of substrings is unclear.

namsengi11 commented on src/main/java/chattpg/logic/TaskOrganiser.java:
```diff
@@ -0,0 +1,166 @@
+                    case "help":
+                        printAvailableCommands();
+                        break;
+                    case "mark done":
```
You could refactor the logic for each case in order to make the run loop more readable.

namsengi11 commented on src/main/java/chattpg/logic/TaskOrganiser.java:
```diff
@@ -0,0 +1,166 @@
+        }
+    }
+
+    public void taskAdded() {
```
Method could be named to better indicate action responsible, such as announceTaskAdded()

namsengi11 commented on src/main/java/chattpg/model/Deadline.java:
```diff
@@ -0,0 +1,17 @@
+
+public class Deadline extends Task {
+
+    private final String by;
```
While I understand why the variable could be named 'by', but it should be more descriptive like 'dueDate'

melchti committed:
Refactored task counting logic to use a dynamically resizeable array

Refactored code to use a dynamically resizeable arraylist instead of a
static array with a maximum task count of only 100. Also added line
separators to improve visuals of the chatbot for aesthetics

melchti committed:
Add method to save tasks and read tasks from a .txt list

Save the tasks in the hard disk automatically whenever the task list changes. Load the data from the hard disk when the chatbot starts up.
File must exist in project root folder, under tasks/tasks.txt for it to
work as the file path is hard-coded.

melchti committed:
Merge branch 'Level-6' into main

melchti committed:
Merge branch 'Level-7' into main. Resolve merge conflicts.

melchti committed:
Merge branch 'Level-7' into main. Add Java Collections classes

Resolve merge conflicts in TaskOrganiser.java, Event.java, Deadline.java

danielkwan2004 committed:
Modify gitignore

melchti committed:
Make the code more OOP

melchti committed:
Add finding function for users to find a task by searching for a keyword in the task description

melchti committed:
Add simple javadoc comments to exceptions

danielkwan2004 committed:
Merge pull request #1 from danielkwan2004/branch-Level-9

Merge branch 'Level-9' into main

melchti committed:
Done javadoc comments for all classes and methods

danielkwan2004 committed:
Merge pull request #2 from danielkwan2004/branch-A-JavaDoc

Merge all javadoc comments for all classes and methods

melchti committed:
Update and publish README.md

melchti committed:
Reformat user guide

melchti committed:
chore: ignore editor and test artifacts

melchti committed:
Modify .gitignore

melchti committed:
Modify other README.md so it is visible on github repo home page

melchti committed:
Fix small UI bug
